amethyst_controls/src/systems.rs

```diff
@@ -106,87 +108,68 @@ where
     B: Send + Sync + Hash + Eq + Clone + 'static,
 {
     type SystemData = (
-        Read<'a, InputHandler<A, B>>,
-        Read<'a, WindowFocus>,
-        ReadExpect<'a, ScreenDimensions>,
+        Read<'a, EventChannel<Event>>,
         WriteStorage<'a, Transform>,
         ReadStorage<'a, FlyControlTag>,
     );
 
-    fn run(&mut self, (input, focus, dim, mut transform, tag): Self::SystemData) {
-        if focus.is_focused {
-            // take the same mid-point as the MouseCenterLockSystem
-            let half_x = dim.width() as i32 / 2;
-            let half_y = dim.height() as i32 / 2;
-
-            if let Some((posx, posy)) = input.mouse_position() {
-                let offset_x = half_x as f32 - posx as f32;
-                let offset_y = half_y as f32 - posy as f32;
-                for (transform, _) in (&mut transform, &tag).join() {
-                    transform.pitch_local(Deg(offset_y * self.sensitivity_y));
-                    transform.yaw_global(Deg(offset_x * self.sensitivity_x));
-                }
-            }
-        }
-    }
-}
-
-/// The system that locks the mouse to the center of the screen. Useful for first person camera.
-pub struct MouseCenterLockSystem;
```

Comment:
Please keep it, it can still be useful too.

Reply:
Can you give an example for when this would be useful over using `Grab`?

Reply:
1) As a fallback from the grab method for other pointer devices.
2) If for some reason you actually want to force the mouse to be centered as a game feature.